stardict.cc

```diff
@@ -466,20 +466,16 @@ string StardictDictionary::handleResource( char type, char const * resource, siz
     {
       QString articleText = QString( "<div class=\"sdct_h\">" ) + QString::fromUtf8( resource, size ) + "</div>";
 
-      const char * srcRegexpStr = "(<\\s*img\\s+[^>]*src\\s*=\\s*[\"']+)(?!(?:data|https?|ftp):)";
-      const char * hrefRegexpStr = "(<\\s*link\\s+[^>]*href\\s*=\\s*[\"']+)(?!(?:data|https?|ftp):)";
+      const char * srcRegexpStr = "(<\\s*(?:img|script)\\s[^>]*src\\s*=\\s*[\"']*)(?![^>]*(?:data|https?|ftp):)";
+      const char * hrefRegexpStr = "(<\\s*link\\s[^>]*href\\s*=\\s*[\"']*)(?![^>]*(?:data|https?|ftp):)";
```

Comment:
Wouldn't replacing `[\"']*` with `[\"']?` (zero or one match) be safer?

Reply:
You are right. `*` was chosen because it may be inner quotes, but we actually do not want to match it.

Reply:
Why don't we want to match inner quotes?

Reply:
Because inner quotes is generally a part of a value, so if it is used for some reason, it should not be truncated from the value anyway.

Reply:
I also do not think we should ignore inner quotes because HTML does not just ignore redundant quotes, or does it?

f29e5b13cb364cf9d194fd81438481d7e07a0912 originally introduced `[\"']*`. Then 85250bb6127734adfe383ef435e48934f92e2e86 introduced the negative look ahead and replaced it with `[\"']+`.

@Abs62, are you OK with not matching more than one quote character here?

Reply:
> I also do not think we should ignore inner quotes because HTML does not just ignore redundant quotes, or does it?

Well, not ignores, but decides as a part of value. [An attribute value may be wrapped with single or double qoutes (and also may be not wrapped at all)](https://html.spec.whatwg.org/multipage/introduction.html#a-quick-introduction-to-html:syntax-attributes). What is inside quotes is a value.